### PR TITLE
MM-1656 Updated file UI

### PR DESCRIPTION
**Pull request by hmhealey**
This is a bigger pull request that does the following:
1) Moves the code that display post attachments out of the PostBody, RootPost, and CommentPost components and into its own FileAttachment and FileAttachmentList components.
2) Updates the components that display post attachments to display additional details about the files.
3) Refactors some of the code around image previews (in the ViewImageModal component)
4) Changes the ViewImageModal to support previewing non-image files and also displays additional info about these files.

Important note: This needs one more change once https://github.com/mattermost/platform/pull/252 is in to replace a couple of references to window.location.origin, but is otherwise is ready for review.

### CODE REVIEW

File: api/file.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ func InitFile(r *mux.Router) {
 
 	sr := r.PathPrefix("/files").Subrouter()
 	sr.Handle("/upload", ApiUserRequired(uploadFile)).Methods("POST")
-	sr.Handle("/get/{channel_id:[A-Za-z0-9]+}/{user_id:[A-Za-z0-9]+}/{filename:([A-Za-z0-9]+/)?.+(\\.[A-Za-z0-9]{3,})?}", ApiAppHandler(getFile)).Methods("GET")
+	sr.Handle("/get/{channel_id:[A-Za-z0-9]+}/{user_id:[A-Za-z0-9]+}/{filename:([A-Za-z0-9]+/)?.+(\\.[A-Za-z0-9]{3,})?}", ApiAppHandler(getFile)).Methods("GET", "HEAD")
```
Comment (coreyhulen) [inline api/file.go]: Do you need to modify the func to handle the head call?  or is golang or gorilla doing some magic and not returning the body?

Comment (hmhealey) [inline api/file.go]: No data is being received the client, but I can add a slight optimization to the server-side code to not even write the file to the response if it's a HEAD request.

**Comment by hmhealey** [comment]
Rebased to support the changes from #252 and added a small optimization based on Corey's question.

**Comment by coreyhulen** [comment]
+1

**Comment by jwilander** [comment]
This has approval from @it33, merging..

**Pull request merged**